tests/ui/useless_conversion.rs

```diff
@@ -41,12 +41,90 @@ fn test_issue_5833() -> Result<(), ()> {
     Ok(())
 }
 
+fn test_issue_5833_alt1() -> Result<(), ()> {
+    let text = "foo\r\nbar\n\nbaz\n";
+    let mut lines = text.lines();
+    if Some("ok") == lines.into_iter().next() {}
+
+    Ok(())
+}
+
+fn test_issue_5833_alt2() -> Result<(), ()> {
+    let text = "foo\r\nbar\n\nbaz\n";
+    let mut lines = text.lines().into_iter();
+    if Some("ok") == lines.next() {}
+
+    Ok(())
+}
+
+fn test_issue_5833_alt3() -> Result<(), ()> {
+    let text = "foo\r\nbar\n\nbaz\n";
+    if Some("ok") == text.lines().into_iter().next() {}
+
+    Ok(())
+}
+
+#[allow(const_item_mutation)]
+fn test_const_into_iter1() -> Result<(), ()> {
+    const NUMBERS: std::ops::Range<i32> = 0..10;
+    let _ = NUMBERS.into_iter().next();
```

Comment:
A small tip for this test and the test above. It's usually helpful to leave a comment in the test indicating if a case should be linted or not. During development, this is often clear, but for the review I had to think about if and why this should be linted. For someone else without this PR as context, understanding this code might be harder.

The way I often choose, is to name the functions/modules accordingly. So `lint_into_iter_on_const_iter` might work or a command like:

```rs
// This should lint, since `NUMBERS` is already an iterator and copy
```

Reply:
Renamed all tests concerned by this change.